Templates design updates

AntoineAugusti opened the pull request:
Make several design changes when creating, editing or sending templates.

## Main changes
- Content changes
- Add more context
- Make it easier to send a test message to yourself
- Clarify how bulk sending works
- Update how we display the scheduler

I'll let @e-pang and @anikbrazeau focus on UI changes and @jimleroyer you can take a quick look at the code!

## Links
- [Figma](https://www.figma.com/file/fRVRKm7cM8MUuBWUJbrfgU/Sending-from-web-interface-%7C-Envoyer-par-interface-web?node-id=2803%3A3756)
- Trello: https://trello.com/c/8HXeIRkx/506-create-a-new-template-update-button-copy-and-hierarchy-create-template-button-goes-to-a-new-screen and https://trello.com/c/N5UK9MXt/507-send-message-update-send-to-add-recipients-add-subheading-description-text-update-positioning-and-copy-of-buttons-and-links

AntoineAugusti commented on app/assets/javascripts/scheduler/SetDateTime/SetDateTime.js:
```diff
@@ -18,46 +18,36 @@ const setDateAndTimeValue = val => {
   }
 };
 
```
Update how the scheduler is displayed. Previously it was appearing when selecting radio buttons, now you can make it appear when clicking a button. If the scheduler is shown and you click "send now", make sure we actually clear the scheduled time

AntoineAugusti commented on app/templates/partials/nav/gc_header.html:
```diff
@@ -51,7 +51,7 @@
     <!-- end sub nav -->
   </div>
-  <div>
+  <div class="pt-5 pb-6">
```
More vertical space around GC Notify or GC Notify + service name

e-pang reviewed (changes requested):
Thanks Antoine, the updates are looking good. 
Just this one update:
- After clicking “Send test message to yourself” the back button goes to the personalise screen instead of back to the previous “Ready to send” screen. Can you update?

anikbrazeau commented on app/translations/csv/fr.csv:
```diff
@@ -1388,4 +1391,17 @@
 "For added security, GC Notify has sent you an email message with a security code to confirm you still control a valid Government email address.","Pour plus de sécurité, GC Notification vous a envoyé un courriel avec un code de sécurité pour confirmer que vous contrôlez toujours une adresse courriel gouvernementale valide."
+"Ready to send?","Prêt à envoyer?"
+"Add recipients","Ajouter des destinataires"
+"Edit this template","Modifier le gabarit"
```
The heading when you click in French becomes "Modifier gabarit" ... is there a way to fix that?

anikbrazeau commented on app/templates/views/send.html:
```diff
@@ -2,28 +2,39 @@
+    )
   }}
 
+  <p class="mb-0">
+    {{ _('Save your file as a') }} {% set csv_acronym = _('Comma Separated Values') %}
+    {% set tsv_acr = _('Tab Separated Values') %} {% set ods_acronym = _('Open Document Spreadsheet') %}
+    <acronym title="{{ csv_acronym }}">{{ _('CSV') }}</acronym>,
+    <acronym title="{{ tsv_acronym }}">{{ _('TSV') }}</acronym>,
+    <acronym title="{{ ods_acronym }}">{{ _('ODS') }}</acronym> {{ _('or Microsoft Excel spreadsheet') }}
+  </p>
+
```
```suggestion
```

anikbrazeau commented on app/templates/views/send.html:
```diff
@@ -2,28 +2,39 @@
-    )}}
+    {{ file_upload( form.file, button_text=_('Choose a file')) }}
   </div>
 
```
```suggestion
  <p class="mb-0">
    {{ _('Save your file as a') }} {% set csv_acronym = _('Comma Separated Values') %}
    {% set tsv_acr = _('Tab Separated Values') %} {% set ods_acronym = _('Open Document Spreadsheet') %}
    <acronym title="{{ csv_acronym }}">{{ _('CSV') }}</acronym>,
    <acronym title="{{ tsv_acronym }}">{{ _('TSV') }}</acronym>,
    <acronym title="{{ ods_acronym }}">{{ _('ODS') }}</acronym> {{ _('or Microsoft Excel spreadsheet') }}
  </p>

```

anikbrazeau commented on app/templates/views/check/ok.html:
```diff
@@ -4,32 +4,30 @@
   <h2 class="heading-medium">{{ _('Recipients') }}</h2>
-  <h3 class="file-name" id="{{ file_contents_header_id }}">{{ original_file_name }}</h3>
+  <p>
+    {{ _("{} recipients from {}.").format(count_of_recipients, original_file_name) }}
```
Also seems to be 2 missing FR strings here?
<img width="636" alt="Screen Shot 2021-05-19 at 4 15 13 PM" src="https://user-images.githubusercontent.com/38330843/118878338-72d7b100-b8bd-11eb-9eaa-0b3bb2123a59.png">

"Review before sending","Réviser avant l'envoi"
"was uploaded.", "a été téléversé"

anikbrazeau commented on app/templates/views/check/ok.html:
```diff
@@ -64,48 +58,44 @@ <h3 class="file-name" id="{{ file_contents_header_id }}">{{ original_file_name }
+  <h2 class="heading-medium">{{ _('Message preview') }}</h2>
+
+  <p>
+    {{ _("The example below previews row {} from {}.").format(preview_row, original_file_name) }}
```
```suggestion
    {{ _("The example below previews one recipient (row {} from {}.").format(preview_row, original_file_name) }})
```

anikbrazeau commented on app/translations/csv/fr.csv:
```diff
@@ -1388,4 +1391,17 @@
+"'{}' template saved","Gabarit '{}' enregistré"
+"Your file must include columns with recipients and variables from your personalised message.","Votre fichier doit inclure des colonnes de destinataires et les variables comprises dans votre message personnalisé."
+"Message preview","Aperçu du message"
+"The example below previews row {} from {}.","L’exemple ci-dessous ne montre que la ligne {} de {}."
```
```suggestion
"The example below previews one recipient (row {} from {}).","L’exemple ci-dessous ne montre qu'un destinataire (de la ligne {} de {})."
```

anikbrazeau commented on app/templates/views/check/ok.html:
```diff
@@ -4,32 +4,30 @@
   <h2 class="heading-medium">{{ _('Recipients') }}</h2>
-  <h3 class="file-name" id="{{ file_contents_header_id }}">{{ original_file_name }}</h3>
+  <p>
+    {{ _("{} recipients from {}.").format(count_of_recipients, original_file_name) }}
```
can this be singular for 1 recipient?

anikbrazeau commented on app/translations/csv/fr.csv:
```diff
@@ -1388,4 +1391,17 @@
+"Your file must include columns with recipients and variables from your personalised message.","Votre fichier doit inclure des colonnes de destinataires et les variables comprises dans votre message personnalisé."
+"Message preview","Aperçu du message"
+"The example below previews row {} from {}.","L’exemple ci-dessous ne montre que la ligne {} de {}."
+"{} recipients from {}.","{} destinataires provenant de {}."
```
can this be singular for 1 recipient?

anikbrazeau commented on app/templates/partials/csv-send.html:
```diff
@@ -1,26 +1,21 @@
 </p>
 
 <div class="spreadsheet" data-module="fullscreen-table">
-  {% set hint= _('Example') %} {% call(item, row_number) list_table( example,
```
Is there a way to translate the example in French? 
"email address" =  "adresse courriel"
"phone number" = "numéro de téléphone"
"example" = "exemple"
<img width="1023" alt="Screen Shot 2021-05-19 at 4 23 48 PM" src="https://user-images.githubusercontent.com/38330843/118879602-ec23d380-b8be-11eb-83e1-a232b64f8bad.png">

AntoineAugusti commented:
@e-pang @anikbrazeau Thanks for your review! Addressed your feedback. Can you check one more time that things are good? @jimleroyer will check stuff on the code side just after.

e-pang reviewed (approved):
Looks good to me now, thanks for updating Antoine!

jimleroyer commented on requirements-app.txt:
```diff
@@ -38,6 +38,6 @@ awscli-cwlogs>=1.4.6,<1.5
 itsdangerous==0.24  # pyup: <1.0.0
 
-git+https://github.com/cds-snc/notifier-utils.git@43.9.1#egg=notifications-utils
+git+https://github.com/cds-snc/notifier-utils.git@789b96d8e015ba7df951723ddb227c03b6d331cc#egg=notifications-utils
```
Do we want to use a tag here instead of the commit SHA ID?

jimleroyer reviewed (approved):
It's one of these big PR where one is tempted to blindly approve. There are  many kind of changes too which does not make this easy. I tried to catch obvious things but it looks good overall.

I left a few comments, some reflecting and others that could be addressed here or later (e.g. using sha id instead of labeled version).

There is a lot in there. This is great work and will have a good impact.

AntoineAugusti commented:
@jimleroyer Thanks for the review!

About the scheduler, I think it's out of scope for this PR. I didn't change a lot of things compared to what we have at the moment and I don't want to start _now_ a refactor of this. I do agree that it could be cleaned up! I'll log a "technical debt" card.

For the utils dependency, I used a commit sha for a short bit on this PR only because it was not merged/tagged just yet and I wanted to be able to pull it in the review app. That was a hack while this PR was in review. I've merged/tagged utils and used a tag before merging 👍
